employees/tests/test_views.py

```diff
@@ -217,3 +212,151 @@ def test_delete_report_view_should_delete_report_on_post(self):
         response = self.client.post(path=self.url)
         self.assertEqual(response.status_code, 302)
         self.assertFalse(Report.objects.filter(pk=self.report.pk).exists())
+
+
+class ReportCustomListTests(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+
+        self.project = ProjectFactory()
+        self.project.members.add(self.user)
+
+        self.report = ReportFactory(author=self.user, project=self.project)
+        self.report.full_clean()
+        self.report.save()
+        self.url = reverse("custom-report-list")
+        self.client.force_login(self.user)
+        self.new_project = ProjectFactory()
+
+    def test_custom_list_view_should_display_users_report_list_on_get(self):
+        response = self.client.get(path=self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.report.description)
+        reports = response.context_data["object_list"]
+        self.assertTrue(self.report in reports)
+
+    def test_custom_list_view_should_not_be_accessible_for_unauthenticated_user(self):
+        self.client.logout()
+        response = self.client.get(path=self.url)
+        self.assertEqual(response.status_code, 302)
+
+    def test_custom_list_view_should_not_display_other_users_reports(self):
+        other_user = UserFactory()
+        other_report = ReportFactory()
+        other_report.full_clean()
+        other_report.save()
+        self.client.force_login(other_user)
+        response = self.client.get(path=self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, self.report.description)
+
+    def test_custom_report_list_view_should_add_new_report_on_post(self):
+        response = self.client.post(
+            path=self.url,
+            data={
+                "date": datetime.datetime.now().date(),
+                "description": "Some description",
+                "project": self.project.pk,
+                "work_hours": "8:00",
+                "task_activities": self.report.task_activities.pk,
+            },
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 2)
+
+    def test_custom_report_list_view_should_not_add_new_report_on_post_if_form_is_invalid(self):
+        response = self.client.post(
+            path=self.url, data={"description": "Some description", "project": self.project.pk, "work_hours": "8:00"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Report.objects.all().count(), 1)
```

Comment:
This looks also a little bit confusing - you check that form is invalid and report should not be created, but here there is some report already. I know this is the one created in `setUp`, but someone not familiar has to dive deep to figure out why. But this is also effect of combining Create and List views together, and tests for them too.

This is not a mistake, but looks a little bit odd.

How about separating tests for create report and listing and joining project to separate classes? This way you could have nice, non-conflicting data created in each `setup` of those test classes. But I leave this up to you, if think it is too much work please leave it as it is, we will then refactor if we decide to split those views.